pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -242,3 +248,31 @@ func panicHandler(handler http.Handler) http.Handler {
 		handler.ServeHTTP(w, r)
 	})
 }
+
+func setDefaultMiddleware(config *configuration.Configuration) (changed bool, err error) {
+	// Default to openshift middleware for relevant types
+	// This allows custom configs based on old default configs to continue to work
+	if config.Middleware == nil {
+		config.Middleware = map[string][]configuration.Middleware{}
+	}
+	for _, middlewareType := range []string{"registry", "repository", "storage"} {
+		if _, exists := config.Middleware[middlewareType]; !exists {
+			config.Middleware[middlewareType] = []configuration.Middleware{}
+		}
+		found := false
+		for _, middleware := range config.Middleware[middlewareType] {
+			if middleware.Name == "openshift" {
+				found = true
+				break
+			}
+		}
+		if found {
+			continue
+		}
+		config.Middleware[middlewareType] = append(config.Middleware[middlewareType], configuration.Middleware{
```

Comment:
> or is our registry unable to run unless all three openshift middlewares are configured?

@liggitt Right now docker-registry will not work if not configured all three middlewares.

> If someone does have custom ones registered, does appending openshift after theirs always make sense?

Appended to this is the right decision here.  If someone created a middleware, it should continue to work.